homeassistant/components/asuswrt/__init__.py

```diff
@@ -77,10 +78,16 @@
         dnsmasq=conf[CONF_DNSMASQ],
     )
 
-    await api.connection.async_connect()
-    if not api.is_connected:
-        _LOGGER.error("Unable to setup component")
-        return False
+    try:
+        await api.connection.async_connect()
+        if not api.is_connected:
+            _LOGGER.error("Unable to setup component")
+            return False
```

Comment:
The is_connect check does not raise the exception and should be moved out to the try-catch block

Reply:
Correction done